src/standard_out_api.rs

```diff
@@ -228,24 +196,24 @@ where
 #[cfg(test)]
 mod tests {
     use super::run_commands_stdout;
-
     use super::ConsoleCommand;
+    use crate::CommandOperations;
     use crate::RestartOptions;
+    use crate::Runner;
 
     #[test]
     fn run_commands() {
-        let commands = vec![
-            ConsoleCommand::full_cmd("test1", "ls -la .").unwrap(),
-            ConsoleCommand::full_cmd("test2", "ls -la ..").unwrap(),
-            ConsoleCommand::full_cmd("test3", "ls -la ../..")
-                .unwrap()
-                .cur_dir(".."),
-        ];
-
-        let mut opts = super::Options::new();
-        opts.restart(RestartOptions::Kill);
-
-        let handle = run_commands_stdout(commands);
+        let handle = run_commands_stdout(
+            Runner::new()
+                .command(ConsoleCommand::full_cmd("test1", "ls -la .").unwrap())
+                .command(ConsoleCommand::full_cmd("test2", "ls -la ..").unwrap())
+                .command(
+                    ConsoleCommand::full_cmd("test3", "ls -la ../..")
+                        .unwrap()
+                        .cur_dir(".."),
+                )
+                .restart(RestartOptions::Kill),
+        );
         let _ = handle.join();
```

Comment:
Another thing I'm wrapping my head around are [documentation tests](https://doc.rust-lang.org/rustdoc/documentation-tests.html). Since you don't perform any assertions here, this could be a candidate for a doctest. It nicely illustrates how to use the API.

Reply:
Yeah, I know docs are missing. I want to do that in another PR.

`cargo test` automatically makes sure that any rust files under `examples` compile safely so for now I'm tempted to not include doctests in the PR